### PR TITLE
Fix panic in sharders

### PR DESCRIPTION
**Pull request by Kishan-Dhakan**
## Fixes
- Panic created by concurrent iteration and write in map

## Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

## Associated PRs (Link as appropriate):
- blobber:
- gosdk:
- system_test:
- zboxcli:
- zwalletcli:
- Other: ...

### CODE REVIEW

**Review by peterlimg — DISMISSED** [review]
@Kishan-Dhakan good catch, please make sure the `UniqueBlockExtensions` map is protected in all the places where it is used. And perhaps make it private and access it through method with lock protection.

**Review by guruhubb — APPROVED** [review]

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | TRUE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | fix/map-panic  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

File: code/go/0chain.net/chaincore/block/entity.go [hunk 1]
```diff
@@ -179,6 +180,13 @@ func NewBlock(chainID datastore.Key, round int64) *Block {
 	return b
 }
 
+func (b *Block) GetUniqueBlockExtensions() (uBlExts map[string]bool) {
+	b.uniqueBlockExtMutex.RLock()
+	defer b.uniqueBlockExtMutex.RUnlock()
+
+	return b.uniqueBlockExtensions
```
Comment (peterlimg) [inline code/go/0chain.net/chaincore/block/entity.go:187]: We need to return the copy of the map, this returning would simply expose the map to public without locks protection.

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | master  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**